### PR TITLE
Add MKL supoprt to auto_mixed_precision.

### PR DESCRIPTION
**Pull request by reedwm**
This extends the auto mixed precision grappler pass to support converting nodes to bfloat16 on MKL-supported CPUs. This functionality is exposed as the `auto_mixed_precision_mkl` grappler pass.

Some Python tests are disabled with MKL because they crash. These crashes are likely not due to the grappler pass itself, but instead due to bugs in MKL-DNN or TensorFlow's MKL integration. Intel has confirmed these crashes do not occur when training real models, so we will fix these after this PR is submitted in followup PRs. Since TensorFlow is not built with MKL by default, the auto_mixed_precision_mkl grappler pass cannot be used except in Intel builds and so these issues do not appear in official TensorFlow releases.

Since TensorFlow is not built with MKL by default, the auto_mixed_precision_mkl grappler pass is not documented.

/CC @nhasabni @nluehr @benbarsdell

Co-authored-by: Niranjan Hasabnis <niranjan.hasabnis@intel.com>

### CODE REVIEW

File: tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h [hunk 1]
```diff
@@ -35,6 +69,35 @@ class AutoMixedPrecisionLists {
     }
   }
 
+  // Subclasses should include these on the ClearList.
+  static void AddTensorListOps(gtl::FlatSet<string>* list) {
+    // Note: if a data structure op (such as TensorListPopBack) is added here,
```
Comment (ezhulenev) [inline tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h:74]: Would go/ifthisthenthatlint work here?

**Comment by nhasabni** [comment]
@googlebot I consent.

**Comment by penpornk** [comment]
@reedwm Maybe you have to post `@googlebot I consent.` too? The bot wants both authors to do this.

**Comment by reedwm** [comment]
@googlebot I consent.

File: tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h (hunk 1 above)
Comment (reedwm) [reply tensorflow/core/grappler/optimizers/auto_mixed_precision_lists.h:74]: Done

**Comment by reedwm** [comment]
Both me and @nhasabni clearly consented, so I'm switching the label to cla:yes. Not sure why @googlebot didn't switch it

**Review by ezhulenev — APPROVED** [review]
